### PR TITLE
Tests for chat.cs

### PR DESCRIPTION
**Pull request by daanl**

### CODE REVIEW

File: JabbR.Test/ChatFacts.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            public IJabbrRepository Repository;
+            public Mock<IConnection> MockedConnection;
+
+            public TestableChat(Mock<IResourceProcessor> mockedResourceProcessor, Mock<IChatService> mockedChatService, IJabbrRepository repository, Mock<IConnection> connection)
+                : base(mockedResourceProcessor.Object, mockedChatService.Object, repository)
+            {
+                this.MockedResourceProcessor = mockedResourceProcessor;
+                this.MockedChatService = mockedChatService;
+                this.Repository = repository;
+                this.MockedConnection = connection;
```
Comment (davidfowl) [inline JabbR.Test/ChatFacts.cs]: Remove "this." and make those properties.

File: JabbR/Models/InMemoryRepository.cs [hunk 2]
```diff
@@ -71,17 +69,17 @@ public void Dispose()
 
         public ChatUser GetUserById(string userId)
         {
-            return _users.FirstOrDefault(u => u.Id.Equals(userId, StringComparison.OrdinalIgnoreCase));
+            return _users.Where(x => x.Id != null).FirstOrDefault(u => u.Id.Equals(userId, StringComparison.OrdinalIgnoreCase));
```
Comment (davidfowl) [inline JabbR/Models/InMemoryRepository.cs]: Why not just an and?

`FirstOrDefault(u => u.Id != null && u.Id.Equals(userId .....))`

File: JabbR/packages.config [hunk 3]
```diff
@@ -16,7 +16,7 @@
   <package id="jQuery.UI.Combined" version="1.8.16" />
   <package id="json2" version="1.0" />
   <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" />
-  <package id="Newtonsoft.Json" version="4.0.4" />
+  <package id="Newtonsoft.Json" version="4.0.5" />
```
Comment (davidfowl) [inline JabbR/packages.config:19]: How come?

Comment (daanl) [inline JabbR/packages.config:19]: Installed Newtonsoft by nuget for Jabbr.Tests, the compile broke cuz it was 4.0.4 so i updated it by nuget. for Jabbr

File: JabbR/Models/InMemoryRepository.cs (hunk 2 above)
Comment (daanl) [inline JabbR/Models/InMemoryRepository.cs]: Fixed now

File: JabbR.Test/ChatFacts.cs (hunk 1 above)
Comment (daanl) [inline JabbR.Test/ChatFacts.cs]: Done

**Pull request merged**